src/odemis/util/img.py

```diff
@@ -607,7 +632,12 @@ def mergeMetadata(current, correction=None):
     if model.MD_PIXEL_SIZE in current:
         pxs = current[model.MD_PIXEL_SIZE]
         pxs_cor = current.get(model.MD_PIXEL_SIZE_COR, (1, 1))
-        current[model.MD_PIXEL_SIZE] = (pxs[0] * pxs_cor[0], pxs[1] * pxs_cor[1])
+        # Do the correction for 2D and 3D
+        if len(pxs) == 2 and len(pxs_cor) == 2:
```

Comment:
I'd recommend to also handle when len(pxs) == 2 and len(pxs_cor) == 3, at least! In such case, just use the first 2.
You probably could write it this way, without any `if`:
```
pxs_cor = current.get(model.MD_PIXEL_SIZE_COR, (1,) * len(pxs))
current[model.MD_PIXEL_SIZE] = zip(p * pc for p, pc in zip(pxs, pxs_cor))
```